data-pipeline-ffi/src/trace_exporter.rs

```diff
@@ -360,11 +360,7 @@ pub unsafe extern "C" fn ddog_trace_exporter_send(
     // necessary that the trace be static for the life of the FFI function call as the caller
     // currently owns the memory.
     //APMSP-1621 - Properly fix this sharp-edge by allocating memory on the Rust side
-    let static_trace: ByteSlice<'static> = std::mem::transmute(trace);
-    match exporter.send(
-        tinybytes::Bytes::from_static(static_trace.as_slice()),
-        trace_count,
-    ) {
+    match exporter.send(trace.as_slice(), trace_count) {
```

Comment:
Minor nit. This should work because ByteSlice [implements Deref](https://github.com/DataDog/libdatadog/blob/3dab0bed2e144ce78c10a2378d1aff8fb5974f7d/ddcommon-ffi/src/slice.rs#L28).  

```suggestion
    match exporter.send(&trace, trace_count) {
```